crates/astria-account-monitor/src/account_monitor.rs

```diff
@@ -0,0 +1,296 @@
+use std::{
+    fmt::{
+        self,
+        Display,
+        Formatter,
+    },
+    time::Duration,
+};
+
+use astria_core::{
+    primitive::v1::Address,
+    protocol::account::v1::AssetBalance,
+};
+use astria_eyre::eyre::{
+    self,
+    WrapErr as _,
+};
+use sequencer_client::SequencerClientExt as _;
+use tokio::time::{
+    interval,
+    timeout,
+};
+use tokio_util::sync::CancellationToken;
+use tracing::{
+    info,
+    instrument,
+    warn,
+};
+
+use crate::{
+    config::{
+        Asset,
+        Config,
+        SequencerAccountsToMonitor,
+    },
+    metrics::Metrics,
+};
+
+pub struct AccountMonitor {
+    shutdown_token: ShutdownHandle,
+    sequencer_abci_client: sequencer_client::HttpClient,
+    sequencer_accounts: SequencerAccountsToMonitor,
+    sequencer_asset: Asset,
+    metrics: &'static Metrics,
+    interval: Duration,
+}
+
+impl AccountMonitor {
+    /// Instantiates a new `Service`.
+    ///
+    /// # Errors
+    ///
+    /// - If the provided `sequencer_abci_endpoint` string cannot be contructed to a cometbft http
+    ///   client.
+    /// - If the provided `sequencer_asset` string cannot be parsed to a valid asset.
+    /// - If the provided `sequencer_accounts` string cannot be parsed to a valid address.
+    /// - If the provided `sequencer_bridge_accounts` string cannot be parsed to a valid address.
+    #[instrument(skip_all, err)]
+    pub fn new(cfg: Config, metrics: &'static Metrics) -> eyre::Result<Self> {
+        let shutdown_handle = ShutdownHandle::new();
+
+        let Config {
+            sequencer_abci_endpoint,
+            sequencer_asset,
+            sequencer_accounts,
+            ..
+        } = cfg;
+
+        let sequencer_cometbft_client =
+            sequencer_client::HttpClient::new(&*sequencer_abci_endpoint).wrap_err_with(|| {
+                format!("failed to create sequencer client for url {sequencer_abci_endpoint}")
+            })?;
+
+        let interval = Duration::from_millis(cfg.query_interval_ms);
+        Ok(Self {
+            shutdown_token: shutdown_handle,
+            sequencer_abci_client: sequencer_cometbft_client,
+            sequencer_accounts,
+            sequencer_asset,
+            metrics,
+            interval,
+        })
+    }
+
+    /// Run the query loop, polling the sequencer for accounts information.
+    ///
+    /// # Errors
+    /// An error is returned if bridge last transaction height is not found.
+    pub async fn run(&self) -> eyre::Result<()> {
+        let Some(_res) = self
```

Comment:
The return value is `()` right? Let's make that explicit. We had cases where we accidentally threw away a return value because we bound it to `_value`:
```suggestion
        let Some(()) = self
```